src/Compatibility/Microsoft.DotNet.ApiSymbolExtensions/Logging/ConsoleLog.cs

```diff
@@ -30,8 +32,20 @@ public virtual void LogWarning(string message) =>
             Console.WriteLine(message);
 
         /// <inheritdoc />
-        public virtual void LogWarning(string code, string message) =>
-            Console.WriteLine($"{code}: {message}");
+        public virtual void LogWarning(string code, string message)
+        {
+            string messageTextWithCode = $"{code}: {message}";
+
+            // Mimic MSBuild which logs suppressed warnings as low importance messages.
+            if (_noWarn.Contains(code))
+            {
+                LogMessage(MessageImportance.Low, messageTextWithCode)
+            }
+            else
+            {
+                Console.WriteLine(messageTextWithCode);
+            }
+        }
```

Comment:
Considering we now know this code can break consumers like it happened with aspnetcore, I think we should have a couple tests to verify the behavior.

Reply:
Agreed. I filed https://github.com/dotnet/sdk/issues/46062 to track adding a new test project for the apicompat global tool.